src/structure/widget.ts

```diff
@@ -648,14 +800,87 @@ export class MoleculeViewer {
         }
     }
 
+    /**
+     * Reset the view by re-centering it and zooming to fit the model as much as
+     * possible inside the vies.
+     *
+     * @param center Should we center on the highlighted central atom? If false,
+     *               the camera center/look-at is the center of the whole
+     *               structure.
+     */
     private _resetView(center: boolean): void {
+        // HACK: orthographic camera is broken in 3Dmol
+        // (https://github.com/3dmol/3Dmol.js/issues/434) so we fake on by using
+        // a perspective camera set far away, with a very narrow field of view.
+        this._viewer.setCameraParameters({ fov: 0.01, z: 2000 });
+
         if (center && this._highlighted !== undefined) {
             this._viewer.zoomTo({ serial: this._highlighted.center });
         } else {
             this._viewer.zoomTo();
         }
-
-        this._viewer.zoom(2);
         this._viewer.setSlab(-1000, 1000);
     }
+
+    /**
+     * Rotate the viewed group so that the given direction (in group
+     * coordinates) is aligned with the z axis (in camera space)
+     *
+     * @param direction axis to align with the camera view
+     */
+    private _viewAlong(direction: [number, number, number]): void {
+        const norm = Math.sqrt(
```

Comment:
Would it be beneficial to use something like https://mathjs.org/docs/reference/functions/norm.html?

Reply:
We don't currently depend on math.js, so we don't have access to this function =)